python/services/dataservice/dmod/dataservice/service.py

```diff
@@ -857,9 +939,20 @@ async def manage_data_provision(self):
                 job.set_status_step(JobExecStep.AWAITING_SCHEDULING)
                 self._job_util.save_job(job)
 
+            # Also, unlink usage for any previously existing, job-based users for which the job is no longer active ...
+            self._unlink_finished_jobs(ds_users=prior_users)
+
             self._job_util.unlock_active_jobs(lock_id)
             await asyncio.sleep(5)
 
+    async def manage_temp_datasets(self):
```

Comment:
I would mark this with no return.

```suggestion
    async def manage_temp_datasets(self) -> typing.NoReturn:
```